central/telemetry/phonehome/interceptor.go

```diff
@@ -60,6 +62,19 @@ func getRequestDetails(ctx context.Context, err error, info *grpc.UnaryServerInf
 		method = "unknown"
 		code = -1
 	}
+
+	if throughGrpcGateway {
+		// By default, all permanent HTTP headers are added grpcgateway- prefix:
+		// https://github.com/grpc-ecosystem/grpc-gateway/blob/master/runtime/mux.go#L106-L114
+		// User-Agent header is occupied with internal grpc-go value:
+		// https://github.com/grpc/grpc-go/blob/0238b6e1cec37b55820b461d3d30652c54efe2c4/clientconn.go#L211-L215
+		userAgentKey := fmt.Sprintf("%suser-agent", runtime.MetadataPrefix)
+		userAgent = strings.Join(ri.Metadata.Get(userAgentKey), ", ")
+	}
+	// Fallback to HTTP
+	if !throughGrpcGateway || userAgent == "" {
+		userAgent = strings.Join(ri.Metadata.Get("User-Agent"), ", ")
```

Comment:
FWIW, with such branching you don't need `userAgent` declaration at L43. If you keep the current structure, pull L43 right before branching and make it `var userAgent string`. However, I'm in favor of Michaël's suggestion below.

Reply:
:heavy_plus_sign: